runtime/cudaq/platform/default/rest/helpers/iqm/iqm.config

```
@@ -16,7 +16,9 @@ GEN_TARGET_BACKEND=true
 LINKLIBS="${LINKLIBS} -lcudaq-rest-qpu"
 
 # Define the lowering pipeline, here we lower to Base QIR
-PLATFORM_LOWERING_CONFIG="expand-measurements,unrolling-pipeline,func.func(lower-to-cfg),canonicalize,func.func(multicontrol-decomposition),iqm-gate-set-mapping"
+# Note: the runtime will dynamically substitute %QPU_ARCH% based on
+# qpu-architecture
+PLATFORM_LOWERING_CONFIG="expand-measurements,unrolling-pipeline,func.func(lower-to-cfg),canonicalize,func.func(multicontrol-decomposition),iqm-gate-set-mapping,func.func(add-dealloc,combine-quantum-alloc),canonicalize,func.func(factor-quantum-alloc),func.func(memtoreg),func.func(qubit-mapping{device=file(%QPU_ARCH%)},delay-measurements),func.func(regtomem),iqm-gate-set-mapping"
```

```suggestion
PLATFORM_LOWERING_CONFIG="expand-measurements,unrolling-pipeline,func.func(lower-to-cfg),canonicalize,func.func(multicontrol-decomposition),iqm-gate-set-mapping,func.func(add-dealloc,combine-quantum-alloc),canonicalize,func.func(factor-quantum-alloc,memtoreg,qubit-mapping{device=file(%QPU_ARCH%)},delay-measurements),func.func(regtomem),iqm-gate-set-mapping"
```

I think canonicalize can be run on functions too. If so:
```suggestion
PLATFORM_LOWERING_CONFIG="expand-measurements,unrolling-pipeline,func.func(lower-to-cfg),canonicalize,func.func(multicontrol-decomposition),iqm-gate-set-mapping,func.func(add-dealloc,combine-quantum-alloc,canonicalize,factor-quantum-alloc,memtoreg,qubit-mapping{device=file(%QPU_ARCH%)},delay-measurements),func.func(regtomem),iqm-gate-set-mapping"
```

---

Thanks, I'll incorporate this change. I think `regtomem` can be lumped into that `func.func()`, too.

---

Yep. Good catch.

---

Updated in ec9ad891